cpp/src/arrow/compute/api_scalar.h

```diff
@@ -117,6 +118,13 @@ class ARROW_EXPORT RoundTemporalOptions : public FunctionOptions {
   CalendarUnit unit;
   /// What day does the week start with (Monday=true, Sunday=false)
   bool week_starts_monday;
+  /// Times exactly on unit multiple boundary will be rounded one unit multiple up.
+  /// This applies for ceiling only.
```

Comment:
So taking the day-ceiling of `2022-01-01 00:00:00` is normally a no-op, but will return `2022-01-02 00:00:00` with this enabled?